db/migrate/20240122064126_create_ads.rb

```diff
@@ -0,0 +1,36 @@
+class CreateAds < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ads, :id => false, primary_key: :ad_id do |t|
+      t.string          :ad_id, null: false
+      t.string          :ad_name
+      t.string          :ad_type
+      t.string          :ad_format
+      t.datetime        :ad_start_date
+      t.string          :ad_set_id, null: false
```

Comment:
this is a foreign key to ad set table, right? don't see foreign key constraint (likewise in other tables where parent is referred)

Reply:
Didn't want to add foreign key constraints, IMO, it would slow down the writes.